File: src/main/java/at/fhhagenberg/sqe/entity/Direction.java
package at.fhhagenberg.sqe.entity;

public enum Direction {

    UP(0),
    DOWN(1),
    UNCOMMITTED(2);

    public final int direction;

    Direction(int direction) {
        this.direction = direction;
    }

    public static Direction valueOf(int direction) {
        for (Direction enumDirection : values()) {
            if (enumDirection.direction == direction) {
                return enumDirection;
            }
        }
        return null;
    }
}

File: src/main/kotlin/at/fhhagenberg/sqe/entity/Direction.kt
package at.fhhagenberg.sqe.entity

import at.fhhagenberg.sqe.util.EnumUtils.valOf

enum class Direction(val direction: Int) {
    UNKNOWN(-1), UP(0), DOWN(1), UNCOMMITTED(2);

    companion object {
        @JvmStatic
        fun valueOf(direction: Int): Direction = valOf { it.direction == direction } ?: UNKNOWN
    }
}